### PR TITLE
Add test case for String#next and String#succ.

### PR DESCRIPTION
**Pull request by K0mAtoru**
Add the test case when the '' and " " of next, next!, succ and succ!

### CODE REVIEW

**Comment by nobu** [comment]
You could re-run CI for #1209 by just `git push -f`.

**Comment by K0mAtoru** [comment]
@nobu 
oh, I get it.
From the next it will do.
Thank you for your teaching :)

File: test/ruby/test_string.rb [hunk 1]
```diff
@@ -1031,6 +1034,12 @@ def test_next!
     a = S("**")
     assert_equal(S("*+"), a.next!)
     assert_equal(S("*+"), a)
+
+    a = S(" ")
+    assert_equal(S("!"), a.next!)
```
Comment (hsbt) [inline test/ruby/test_string.rb]: `next!` is destructive method. Please add assertion of `a` value same as L1033.

File: test/ruby/test_string.rb [hunk 2]
```diff
@@ -1582,6 +1594,12 @@ def test_succ!
     assert_equal(S("No.10"), a.succ!)
     assert_equal(S("No.10"), a)
 
+    a = S(" ")
+    assert_equal(S("!"), a.succ!)
```
Comment (hsbt) [inline test/ruby/test_string.rb]: Same here with https://github.com/ruby/ruby/pull/1213/files#r50620274

**Comment by K0mAtoru** [comment]
@hsbt 
Thank you for your teaching.
I fixed your poinig out.

**Pull request closed without merging**